fabric-object-builders-v0/src/main/java/net/fabricmc/fabric/api/world/poi/v1/PointOfInterestHelper.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.world.poi.v1;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+import com.google.common.collect.ImmutableSet;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.poi.PointOfInterestType;
+
+import net.fabricmc.fabric.mixin.object.builder.PointOfInterestTypeAccessor;
+
+/**
+ * This class provides utilities to create a {@link PointOfInterestType}.
+ */
+public interface PointOfInterestHelper {
+	/**
+	 * Registers and creates a {@link PointOfInterestType}.
+	 * @param workStationStates A list of {@link BlockState BlockStates} which this {@link PointOfInterestType} can be present on.
+	 * @param id The id of this {@link PointOfInterestType}.
+	 * @param ticketCount The amount of tickets.
+	 * @param searchDistance The search distance.
+	 * @return a new {@link PointOfInterestType}.
+	 */
+	static PointOfInterestType create(Set<BlockState> workStationStates, Identifier id, int ticketCount, int searchDistance) {
+		return PointOfInterestTypeAccessor.callSetup(Registry.POINT_OF_INTEREST_TYPE.add(id, PointOfInterestTypeAccessor.callCreate(id.toString(), workStationStates, ticketCount, searchDistance)));
```

Comment:
Call `Registry.register` here instead maybe? Not sure which is better to do.

Reply:
Arguably I could do either or make the registry call not part of this method, the setup basically binds all the block states to the point of interest type and verifies two pois don't share a state.